
<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta http-equiv="cache-control" content="max-age=0" />
<meta http-equiv="cache-control" content="no-cache" />
<meta http-equiv="expires" content="0" />
<meta http-equiv="expires" content="Tue, 01 Jan 1980 1:00:00 GMT" />
<meta http-equiv="pragma" content="no-cache" />
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />

<link rel="stylesheet" type="text/css" href="style.css">

<title>Yongchao Liu@Ant Financial</title>

</head>

<body>
<!--header-->
<div id="header"></div>

<!--side bar-->
<div id="sidebar">

<!--photo-->
<div id="profile"></div>
<p style="text-align:center">[<a href="contact.html">Contact</a>][<a href="https://scholar.google.com/citations?user=qYQHl4sAAAAJ&hl=en&oi=ao">Google Scholar</a>]</p>

<!--navigator-->
<div id="nav">
<ul>
<li><a href="index.html">About Me</a></li>
<li><a href="news.html">What's New</a></li>
<li><a href="awards.html">Awards</a></li>
<li><a href="grants.html">Grants</a></li>
<li><a href="publications.html">Publications</a></li>
<li><a href="software.html">Software</a></li>
<li><a href="projects.html">Projects</a></li>
<li><a href="services.html">Services</a></li>
<li><a href="talks.html">Talks</a></li>
<li><a href="teaching.html">Teaching</a></li>
<li><a href="people.html">People</a></li>
<li><a href="contact.html">Contact</a></li>
</ul>
</div>

</div><!--sidebar-->

<div id="main">

<h1>Brief Bio</h1>
<hr>

<!--This part is for brief biography-->
<p><b>Dr. Yongchao Liu</b> is currently a Staff Engineer in <a href="https://www.antfin.com">Ant Financial</a> (from 07/2017 to present), working on distributed machine learning (online and graph learning) and heterogeneous computing platforms for deep learning (neural network compressors and compilers). Prior to that, he worked as a Research Scientist II (a research faculty member) in the School of Computational Science &amp; Engineering, <a href="http://www.cse.gatech.edu">Georgia Institute of Technology (USA)</a> (from 01/2015 to 07/2017), and as a Postdoctoral Researcher in the Institute of Computer Science, <a href="http://www.informatik.uni-mainz.de/">University of Mainz (Germany)</a> (from 11/2011 to 01/2015). He earned his Ph.D degree in computer engineering from <a href="http://www.chipes.ntu.edu.sg/">Nanyang Technological University (Singapore)</a> in 2012 (supervised by <a href="http://www.staff.uni-mainz.de/schmi033">Dr. Bertil Schmidt</a> and <a href="http://www.ntu.edu.sg/home/asdouglas/">Dr. Douglas Maskell</a>). Prior to that, he earned the Master and Bachelor degrees in computer science and technology from <a href="http://english.nankai.edu.cn">Nankai University (China)</a> in 2008 and 2005, respectively.
</p>

<p>He is an internationally recognized researcher in parallel computing and bioinformatics. He demonstrated outstanding contributions to scientific communities by his novel parallel algorithms and software tools for large-scale biological data analysis. These parallel algorithms and tools streamline fundamental and computationally challenging biological issues with parallel computing, via in-depth exploration of current high performance computing techniques and technologies such as hardware accelerators (e.g. GPUs and Intel Xeon Phis) and clusters (e.g. CPU/GPU/Xeon Phi clusters). On the other hand, these algorithms and tools investigate a set of related critical issues in bioinformatics and have actually established an innovative and unique analysis platform for large-scale biological datasets, especially next-generation sequencing reads. To address big data challenges, he proposed a new concept of computing, i.e. <a href="https://www.researchgate.net/publication/309672981_Research_Statement_for_Assistant_Professorship_Application_2016">Compact Computing</a>, which centers around data and enables full-stack computation by tightly coupling algorithms with systems. His research interests focus on parallel and distributed algorithm design for bioinformatics, heterogeneous computing with accelerators (GPUs and Xeon Phis), high performance computing on big data, and parallelized machine learning.</p>

<p>He has released a set of software tools for reproducible research and public use, most of which are open-source, associated with his paper publications. Among these algorithms, three CUDA-based open-source software tools, i.e. <a href="http://cudasw.sourceforge.net/">CUDASW++</a>, <a href="http://cuda-meme.sourceforge.net">mCUDA-MEME</a> and <a href="http://cushaw.sourceforge.net/">CUSHAW</a>, are rated by NVIDIA Corporation as <a href="http://www.nvidia.com/docs/IO/123576/nv-applications-catalog-lowres.pdf">popular GPU-accelerated applications</a>, while <a href="http://decgpu.sourceforge.net">DecGPU</a> (the first parallel and distributed error correction algorithm for high-throughput short reads) was reported by <a href="data/next_gen_gpus.pdf">GenomeWeb</a>. Furthermore, some of his other open-source tools are also leading in their respective areas, including <a href="http://masprobs.sourceforge.net">MSAProbs</a> (multiple protein sequence alignment), <a href="http://musket.sourceforge.net">Musket</a> (Illumina reads error correction), <a href="http://cushaw2.sourceforge.net">CUSHAW2</a> (NGS base-space read alignment), <a href="http://cushaw3.sourceforge.net">CUSHAW3</a> (NGS base-space and color-space read alignment), <a href="http://pasha.sourceforge.net">PASHA</a> (NGS de novo genome assembly), <a href="http://snvsniffer.sourceforge.net">SNVSniffer</a> (NGS germline and somatic SNV calling), <a href="http://swaphi.sourceforge.net">SWAPHI</a> (Xeon-Phi-based protein database search), <a href="http://swaphi-ls.sourceforge.net">SWAPHI-LS</a> (Xeon-Phi-based pairwise DNA sequence alignment), <a href="http://parabwt.sourceforge.net">ParaBWT</a> (parallel construction of Burrows-Wheeler transform and suffix array), <a href="http://lightspmv.sourceforge.net">LightSpMV</a> (CUDA-based sparse matrix-vector multiplication), and <a href="http://lightpcc.sourceforge.net">LightPCC</a> (parallel correlation computation on Xeon Phi clusters).</p>

<p>He won two Best Paper Awards from IEEE International Conference on Application-specific Systems, Architectures and Processors (IEEE ASAP) in 2009 and 2015, got one paper recommended for Best Paper Award from IEEE International Conference on Cluster Computing (IEEE Cluster) in 2014, won the Program to Empower Partnerships with Industry Award from U.S. South Big Data Hub in 2016 and Annual Best Paper Award of Ant Group in 2021. He led his team to have won the Championship on Stanford University Open Graph Benchmark proteins dataset leaderboard in 2021. Moreover, the landing of his large-scale distributed graph learning system on Zhima Credit business was listed in Ant Technology Memorabilia in 2020. He was listed in the reputable Who's Who in America (Marquis research) in 2016. He was awarded Innovative Talent (in artificial intelligence) of Hangzhou 521 Program for Global Talents Introduction in 2019 and was recognized as Hangzhou High-Level Talents of Category C in 2020. He co-chaired the first <a href="http://psalsa.gatech.edu">Workshop on Parallel Software Libraries for Sequence Analysis (pSALSA)</a> in 2015 and founded the <a href="wacebi2016">Workshop on Accelerator-Enabled Algorithms and Applications in Bioinformatics (WACEBI)</a> in 2016. In addition, he serves as a reviewer for some top journals, such as Nature Methods, Nature Communications, Bioinformatics, TPDS and TCBB, and as a program committee member for some leading conferences such as IPDPS and CCGrid.</p>
 
<br>
 <h3><font color="red">Interesting news</font></h3>
  <ul>
    <li> Our paper <a href="", style="text-decoration:none">Heterogeneous graph neural architecture search</a> got accepted by ICDM 2021, Auckland, New Zealand.</li>
    <li>Our paper "<a href="https://2021.ecmlpkdd.org/wp-content/uploads/2021/07/sub_1040.pdf" style="text-decoration: none">Adaptive optimizers with sparse group lasso for neural networks in CTR prediction</a>" gets accepted by ECML-PKDD 2021.</li>
    <li>Won the championship on the Stanford University Open Graph Benchmark proteins dataset leaderboard: https://ogb.stanford.edu/docs/leader_nodeprop/#ogbn-proteins</li>
    <li>Our graph learning system paper won the Annual Best Paper Award (1st place) in Ant Group, 2021</li>
    <li>Our graph learning system was listed in 2020 Ant Technology Memorabilia (2020年蚂蚁技术大事记), 2021</li>
    <li>Our paper "Path-based Deep Network for Candidate Item Matching in Recommenders“ has got accepted by SIGIR 2021 as a full-paper.</li>
    <li><a href="ai_recycle.html">AI Smart Phone Recycling Robot (AI智能手机回收机器人)</a></li>
    <li>Our preliminary research on learning-to-compiling for deep learning is avaiable now: <a href="https://arxiv.org/pdf/2008.04567.pdf">Woodpecker-DL: Accelerating Deep Neural Networks via Hardware-Aware Multifaceted Optimizations</a>. 08/2020.</li>
  </ul>
<br>
<hr>
<div>
<table align="center" cellpadding="0">
<tr>
<td align="center">
<font color="green"><i>Curiosity of the world motivates me to experience diverse styles of life!</i></font>
</td>
<td>
<a href="http://www.antfin.com"> <img src="icons/antfinancial.jpg" width="130"></a></td>
<td>
<a href="http://www.gatech.edu"> <img src="icons/gatech.jpg" width="130"></a></td>
<td>
<a href="http://uni-mainz.de"><img src="icons/jgu.png" width="130"></a></td>
<td>
<a href="http://www.ntu.edu.sg"><img src="icons/ntu.jpg" width="130"></a></td>
<td>
<a href="http://www.nku.cn"><img src="icons/nku.jpg" width="80"></a></td>
</td>
<td>
<a href="http://yongchao-liu.github.io"><img src="icons/liuy.png" width="80"></a></td>
</tr>
</table>
</div><!--footer-->

</div><!--main-->
</body>
</html>
