<html><head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta http-equiv="cache-control" content="max-age=0" />
<meta http-equiv="cache-control" content="no-cache" />
<meta http-equiv="expires" content="0" />
<meta http-equiv="expires" content="Tue, 01 Jan 1980 1:00:00 GMT" />
<meta http-equiv="pragma" content="no-cache" />
<link rel="stylesheet" type="text/css" href="style.css">
	<title>What is New!</title>
</head>

<!--header-->
<div id="header"></div>

<!--side bar-->
<div id="sidebar">

<!--photo-->
<div id="profile"></div>
<p style="text-align:center">[<a href="contact.html">Contact</a>][<a href="https://scholar.google.com/citations?user=qYQHl4sAAAAJ&hl=en&oi=ao">Google Scholar</a>]</p>

<!--navigator-->
<div id="nav">
<ul>
<li><a href="index.html">About Me</a></li>
<li><a href="news.html">What's New</a></li>
<li><a href="awards.html">Awards</a></li>
<li><a href="grants.html">Grants</a></li>
<li><a href="publications.html">Publications</a></li>
<li><a href="software.html">Software</a></li>
<li><a href="projects.html">Projects</a></li>
<li><a href="services.html">Services</a></li>
<li><a href="talks.html">Talks</a></li>
<li><a href="teaching.html">Teaching</a></li>
<li><a href="people.html">People</a></li>
<li><a href="contact.html">Contact</a></li>
</ul>
</div>

<!--last update-->
<p class="foot">
	<script language="JavaScript">
		document.write("Updated: " + document.lastModified);
	</script>
</p>

</div><!--sidebar-->

<div id="main">

<h1>What's New</h1>
<hr>
<h4>2020</h4>
<ul>
<li>[<b>April 30, 2020</b>] My first U.S. patent has got granted: "<a href="https://patents.google.com/patent/US20200134400A1/en">Fast computation of a convolutional neural network</a>".</li>
</ul>
	
<h4>2019</h4>
<ul>
<li>[<b>December 31, 2019</b>] Awarded Hangzhou 521 Program for Global Talents Introduction: Innovative Talent in Artificial intelligence</li>
<li>[<b>November 4, 2019</b>] We will present "<i>Woodpecker-DL: an efficient compiler for accelerating deep learning
	on heterogeneous computing architectures (Woodpecker-DL: 一个基于异构体系结构加速深度学习的高效编译器)</i>" at <a href="https://www.nvidia.cn/gtc/topics/developer-tools/">GPU Technology Conference 2019 (GTC China 2019)</a>, Suzhou, China. </li>
<li>[<b>September 10, 2019</b>] Our work "<i>Woodpecker-AutoSearch: a machine learning-based framework for math kernel optimizations</i>" will be presented at the DAWN retreat by Stanford University, 2019.</li>
</ul>

<h4>2017</h4>
<ul>
<li>[<b>July 06, 2017</b>] I started working at Ant Financial as a senior technical expert to develop parallel machine learning algorithms and heterogeneous computing systems/platforms for artificial intelligence finance.</li>
	
<li>[<b>March 09, 2017</b>] delivered an invited talk at <b>Binghamton University</b> (State University of New York at Binghamton), Binghamton, USA, 03/2017</li>

<li>[<b>January 20, 2017</b>] delivered an invited talk at <b>Fairfield University</b>, Fairfield, USA, 01/2017</li>

<li>[<b>January 13, 2017</b>] delivered an invited talk at Seven Bridges Genomics (a biomedical data analysis company accelerating breakthroughs in genomics research for cancer, drug development and precision medicine), Boston, USA</li>
</ul>

<h4>2016</h4>
<ul>

<li>[<b>December 20, 2016</b>] Our extended <a href="http://lightspmv.sourceforge.net">LightSpMV</a> paper, entitled "LightSpMV: faster CUDA-compatible sparse matrix-vector multiplication using compressed sparse rows", gets accepted by Journal of Signal Processing Systems.</li>

<li>[<b>November 10, 2016</b>] I will be officialy listed in the 71st edition of <a href="http://www.cc.gatech.edu/~yliu/data/whos_who_in_america.pdf">Who's Who in America</a>. </li>

<li>[<b>November 04, 2016</b>] I released my Research Statement for Assistant Professorship Application (2016) publicly in ResearchGate. I believe that the themes and topics that I am planning to investigate in the future could be interesting to you! Click <a href="https://www.researchgate.net/publication/309672981_Research_Statement_for_Assistant_Professorship_Application_2016">here</a> to visist the research statement and welcome any feedback!</li>

<li>[<b>October 24, 2016</b>] I got nominated to be listed in the 71st edition of <a href="https://www.marquiswhoswho.com">Who's Who in America</a>, for which Marquis states in the Preface that <i>Who's Who in America</i> "endeavors to profile the leaders of American society; those men and women who are influencing their nation's development" [<a href="https://en.wikipedia.org/wiki/Marquis_Who%27s_Who">Wiki</a>].</li>

<li>[<b>August 30, 2016</b>] I released my first <a href="https://www.researchgate.net/publication/309673143_OpenGraphAssembly_abstract_modularize_and_parallelize_fundamental_building_blocks_for_graph-based_genome_assembly">OpenGraphAssembly</a> white paper that was originally submitted to apply for Research Scientist Development Grants of Georgia Tech in June 2015. I will release more and more white papers in the future and share my thoughts with researchers from related community.</li>

<li>[<b>August 22, 2016</b>] Our MSAProbs-MPI paper gets accepted by the Bioinformatics journal. The software is publicly available at <a href="http://msaprobs.sourceforge.net">here</a>.</li>

<li>[<b>August 14, 2016</b>] Our paper "Parallel pairwise correlation computation On Intel Xeon Phi clusters" get accepted by <b>SBAC-PAC 2016</b> conference. The corresponding software <i>LightPCC</i> is available at <a href="http://lightpcc.sourceforge.net">here</a>. </li>

<li>[<b>August 1, 2016</b>] Our SNVSniffer 2.0 paper gets published <a href="http://bmcsystbiol.biomedcentral.com/articles/10.1186/s12918-016-0300-5">online</a> in the BMC Systems Biology paper.</li>

<li>[<b>July 17, 2016</b>] Our "Kmerind: A flexible parallel library for k-mer indexing of biological sequences on distributed memory systems" paper gets accepted by ACM-BCB 2016.</li>

<li>[<b>May 26, 2016</b>] NVIDIA has agreed to sponsor a Best GPU Paper/Talk Award for our <a href="http://www.cc.gatech.edu/~yliu/wacebi2016">WACEBI 2016</a> workshop. They will donate a GeForce GTX Titan X graphics card as the prize. Welcome to submit your high-quality papers to our workshop!</li>

<li>[<b>May 18, 2016</b>] Our paper entitled"SNVSniffer: an integrated caller for germline and somatic single-nucleotide and indel mutations" get accepted by BMC Systems Biology. The software tool is publicly available at <a href="http://snvsniffer.sourceforge.net">here</a>. </li>

<li>[<b>May 13, 2016</b>] Awarded The Program to Empower Partnerships with Industry (PEPI) Award from the South Big Data Regional Innovation Hub, USA, 2016.</li>

<li>[<b>May 05, 2016</b>] The pre-print of our paper "<a href="https://arxiv.org/abs/1605.01584">Parallel Pairwise Correlation Computation On Intel Xeon Phi Clusters</a>" is made available at arXiv for public review before formal submission to a journal/conference.</li>

<li> [<b>May 02, 2016</b>] <a href="http://msaprobs.sourceforge.net" rel="nofollow">MSAProbs-MPI</a> gets released online. By using distributed-memory systems, we manage to overcome high memory overhead barriers for multiple alignment of thousands of protein sequences. By scaling with hundreds of cores, we can reach faster speed. </li>

<li> [<b>April 17, 2016</b>] Because of the very long peer-review time, I decided to make the <a href="http://arxiv.org/abs/1604.04815" rel="nofollow">LightScan (parallel scan primitives on CUDA-enabled GPUs) preprint</a> publicly available at arXiv. Welcome any feed about this pre-print!</li>

<li> [<b>March 20, 2016</b>] As the first news of the year, our <a href="http://snvsniffer.sourceforge.net">SNVSniffer 2.0</a> gets released today! The paper has been submitted to a journal for peer review. Welcome any feed back about this software! Currently, only statically linked executable binary is available and the source code will be released some time later.</li>
</ul>

<h4>2015</h4>
<ul>

<li> [<b>December 29, 2015</b>] Our SNVSniffer paper for germline and somatic variant calling has got published <a href="http://dx.doi.org/10.1109/BIBM.2015.7359659">online</a>.</li>

<li>[<b>December 07, 2015</b>] Our <a href="http://cushaw3.sourceforge.net">CUSHAW3-UPC</a> paper has got accepted by PLOS One.</li>

<li>[<b>November 20, 2015</b>] We are happy to see that our <a href="http://musket.sourceforge.net">Musket</a> error correction algoirthm is used to assemble and annotate the genome of the UCSC mascot Ariolimax dolichophallus (i.e. the slender banana slug). Refer to the bottom of <a href="https://banana-slug.soe.ucsc.edu/data_overview:2015:ucsf_sw019">the page</a>.</li>

<li>[<b>November 06, 2015</b>] Our paper entitled "<a href="http://dx.doi.org/10.1016/j.parco.2015.11.001">Bit-parallel approximate pattern matching: Kepler GPU versus Xeon Phi</a>" has been published online.</li>

<li>[<b>November, 02, 2015</b>] Our GPU-accelerated Wu-Manber algorithm gets accepted by the Parallel Computing journal.</li>

<li>[<b>October 09, 2015</b>] The new <a href="https://sites.google.com/site/yongchaosoftware/mcuda-meme">CUDA-MEME 3.0.16</a> gets released, where we have fixed a bug that causes memory segmentation when using multiple threads for the EM stage.</li>

<li>[<b>September 13, 2015</b>] Our open-source software <a href="http://cushaw3.sourceforge.net">CUSHAW3-UPC</a> has got released. This software is a parallel and scalable NGS short-read aligner using the UPC++ parallel programming language and has been extensively tested on a compute cluster with 2048 CPU cores. Welcome any suggestions and comments! </li>

<li>[<b>September 08, 2015</b>] Awarded the NVIDIA Academic hardware grant that donates one GeForce GTX TITAN X graphics card to support my research.</li>

<li>[<b>September 07, 2015</b>] Our <a href="snvsniffer.sourceforge.net">SNVSniffer</a> paper for the calling of SNPs and somatic variants gets accepted by <a href="http://cci.drexel.edu/ieeebibm/bibm2015/">IEEE BIBM 2015</a>.</li>

<li>[<b>September 01, 2015</b>] One paper gets accepted by the <a href="http://www.iccabs.org/">IEEE ICCABS 2015</a> conference.</li>

<li>[<b>July 29, 2015</b>] Our <a href="http://lightspmv.sourceforge.net" rel="nofollow">LightSpMV</a> paper just won the <font style="color:rgb(255,0,0)">Best Paper Award</font> from the prestigious IEEE ASAP 2015 conference</li>

<li>[<b>May 16, 2015</b>] Our CUDA-based <a href="http://lightspmv.sourceforge.net">LightSpMV</a> paper for <b>Sparse Matrix-Vector Multiplication</b> has been accepted by IEEE ASAP 2015.</li>

<li>[<b>May 08, 2015</b>] Our ParaBWT paper has been published <a href="http://dx.doi.org/10.1109/TCBB.2015.2430314">online</a>. 
</li>
<li>[<b>April 22, 2015</b>] Our paper "<a href="http://parabwt.sourceforge.net">Parallel and Space-efficient Construction of Burrows-Wheeler Transform and Suffix Array for Big Genome Data</a>" has been accepted by IEEE TCBB.

</li>
<li>[<b>March 25, 2015</b>] We have removed the dependency on CUDA SDK in <a href="http://cudasw.sourceforge.net">CUDASW++</a>.

</li>
<li>[<b>March 16, 2015</b>] Our poster, named <a href="http://on-demand.gputechconf.com/gtc/2015/posters/GTC_2015_Developer_Algorithms_11_P5267_WEB.pdf">"Faster Compressed Sparse Row (CSR)-based Sparse Matrix-Vector Multiplication using CUDA"</a>, has been published online in <b>GTC 2015</b>.</li>

<li>[<b>Feburary 09, 2015</b>] The source code of our <a href="http://swaphi.sourceforge.net">SWAPHI</a> software has been released!</li>

<li>[<b>Feburary 05, 2015</b>] Get invited to act as a PC member of <a href="http://www.hipc.org/hipc2015/index.php">HiPC 2015</a> and <a href="http://pbc.ppam.pl/">PBC 2015</a>.</li>

<li>[<b>January 23, 2015</b>] Our GPU-based sparse matrix-vector multiplication (SpMV) algorithm has been accepted to present in GPU Technology Conference (GTC) 2015 as a poster. click <a href="http://lightspmv.sourceforge.net/posters/GTC2015poster.pdf">here</a> to download the poster.</li>
<li>[<b>January 19, 2015</b>] Start working as a Research Scientist II (research faculty member) at Georgia Institute of Technology (USA).
</ul>

<h4>2014</h4>
<ul>
</li>
<li>[<b>December 09, 2014</b>] Our paper <a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=6970650">
Bit-Parallel Approximate Pattern Matching on the Xeon Phi Coprocessor</a> has been published online.

</li>
<li>[<b>December 03, 2014</b>] Our paper <a href="http://dx.doi.org/10.1109/CLUSTER.2014.6968772">SWAPHI-LS: Smith-Waterman Algorithm on Xeon Phi coprocessors for Long DNA Sequences</a> has been published online.

</li>
<li>[<b>November 05, 2014</b>] An improved version of <a href="http://snvsniffer.sourceforge.net">SNVSniffer (v1.0.19)</a> has been released. In this new version, we have introduced a new algorithm to estimate the purity of the tumor sample. This purity estimation algorithm works well on simulated normal-tumor samples through our evaluations. </li>
<li>[<b>October 04, 2014</b>] Get invited to serve as a PC member for <a href="http://www.ipdps.org">IPDPS 2015</a>.</li>

<li>[<b>September 23, 2014</b>] The first version of <a href="http://snvsniffer.sourceforge.net">SNVSniffer</a>, an integrated caller for germline and somatic SNVs, is online for public use and test now.</li>

<li>[<b>September 10, 2014]</b> Our GSWABE paper has been published <a href="http://onlinelibrary.wiley.com/doi/10.1002/cpe.3371/full">online</a>. </li>


<li>[<b>August 21, 2014</b>] An invited paper, named "GSWABE: faster GPU-accelerated sequence alignment with optimal alignment retrieval for short DNA sequences",&nbsp; has just been accepted by Concurrency and Computation: Practice and Experience (impact factor 0.784).</li>


<li>[<b>July 31, 2014</b>] Our All-Food-Seq paper has got published <a href="http://www.biomedcentral.com/1471-2164/15/639/abstract">online</a> at BMC Genomics.</li>

<li>[<b>July 24, 2014</b>] Our All-Food-Seq (AFS) paper has just been accepted by <b>BMC Genomics</b> (impact factor 4.40).</li>

<li>[<b>June 30, 2014</b>] The paper "SWAPHI-LS: Smith-Waterman Algorithm on Xeon Phi Coprocessors for Long DNA Sequences" has been <span style="color:rgb(204,0,0)"><b>accepted by IEEE Cluster 2014</b></span>, and has also been <font color="#cc0000" style="font-weight:bold">recommended for Best Paper Award</font>. Click <a href="http://swaphi-ls.sourceforge.net">here </a>to download the open-source SWAPHI-LS software.</li>

<li>[<b>May 15, 2014</b>] The paper "<a href="http://link.springer.com/chapter/10.1007%2F978-3-642-55195-6_23">Faster GPU-Accelerated Smith-Waterman Algorithm with Alignment Backtracking for Short DNA Sequences</a>" has got published online.</li>


<li>[<b>May 07, 2014</b>] Our <a href="http://www.biomedcentral.com/1471-2105/15/131/abstract">HECTOR</a> paper gets published online by the BMC Bioinformatics journal.</li>

<li>[<b>April 26,2014</b>] One solution to compile MSAProbs&nbsp;on <b>Mac OS X</b> has been given. Please refer to the <a href="http://msaprobs.sourceforge.net">project homepage</a> for more details.</li>


<li>[<b>April 24, 2014</b>] One co-authored paper, named "HECTOR: A parallel multistage homopolymer spectrum based error corrector for 454 sequencing data", gets accepted by the BMC Bioinformatics journal. The open-source software can be obtained from <a href="http://hector454.sourceforge.net/">here</a>.</li>


<li>[<b>April 23, 2014</b>] I got invited to join the Program Committee of <a href="http://2014.cloudcom.org">IEEE CloudCom 2014</a>.</li>

<li>[<b>April 16, 2014</b>] The original submission of the <a href="http://swaphi.sourceforge.net/">SWAPHI</a> paper, to the IEEE ASAP 2014 conference, can be obtained from <a href="http://arxiv.org/abs/1404.4152">arXiv</a>. </li>

<li>[<b>April 11, 2014</b>] The <a href="http://swaphi.sourceforge.net">SWAPHI </a>paper has got accepted by <a href="http://www.zurich.ibm.com/asap2014">ASAP 2014</a> as a 2-page short paper.</li>

<li>[<b>February 20, 2014</b>] The <a href="http://on-demand.gputechconf.com/gtc/2014/poster/pdf/P4117_CUSHAW_sequencing_CUDA_GPU.pdf">CUSHAW software package (poster)</a> will be presented in GPU Technology Conference 2014.</li>


<li>[<b>January 24, 2014</b>] The <a href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0086869;jsessionid=5B52DB62469FFD67FAAB3BC2E9C9A3E3">CUSHAW3 paper</a> has been published online.</li>
</ul>

<h4>2013</h4>
<ul>
<li>[<b>December 17, 2013</b>] The <a href="http://cushaw3.sourceforge.net" target="_blank">CUSHAW3</a> paper, named "CUSHAW3: sensitive and accurate base-space and color-space short-read alignment with hybrid seeding", has got accepted by the PLOS ONE journal.</li>


<li>[<b>November 20, 2013</b>] Have submitted my first DFG proposal as a principal investigator (PI). Took me about one month to finish the writing! EXCITED!</li>


<li>[<b>November 04, 2013</b>] A book chapter named "<a href="http://link.springer.com/protocol/10.1007/978-1-62703-646-7_14">Multiple protein sequence alignment with MSAProbs</a>" gets published in the Methods in Molecular Biology book series (Springer).</li>

<li>[<b>October 11, 2013</b>] Just fixed a bug of <a href="http://pasha.sourceforge.net" target="_blank">PASHA</a>, which results in out-of-bound array accesses.</li>


<li>[<b>October 10, 2013</b>] The <a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?tp=&amp;arnumber=6617698&amp;queryText%3DCUSHAW2-GPU%3A+empowering+faster+gapped+short-read+alignment+using+GPU+computing" target="_blank">CUSHAW2-GPU</a> paper has already be published online.</li>


<li>[<b>October 05, 2013</b>] The source code of <a href="http://cushaw3.sourceforge.net" target="_blank">CUSHAW3 </a>has got released.</li>

<li>[<b>August 12, 2013</b>] The <a href="http://swaphi.sourceforge.net">SWAPHI (v1.0.4) </a>software gets first released, which is the first parallelized Smith-Waterman protein database search algorithm on <b><i>Intel Xon Phi coprocessors</i></b>. The corresponding paper has been submitted.</li>


<li>[<b>July 23, 2013</b>] The <a href="http://cushaw2.sourceforge.net" target="_blank">CUSHAW2-GPU</a> paper has been accepted by IEEE Design &amp; Test of Computers (entitled "CUSHAW2-GPU: empowering faster gapped short-read alignment using GPU computing").</li>


<li>[<b>June 22, 2013</b>] The paper, entitled "Faster GPU-accelerated Smith-Waterman algorithm with alignment backtracking for short DNA sequences", has been accepted by the PBC 2013 workshop (in conjunction with PPAM 2013 conference in Poland).</li>

<li>[<b>June 20, 2013</b>] A new version of <a href="http://pasha.sourceforge.net/">PASHA (v1.0.9)</a> got released. In this release, some changes have been made to significantly reduce the mis-assembly rate in <i>scaffolds</i>.</li>


<li>[<b>June 13, 2013</b>] Just given an invited talk about <a href="http://pasha.sourceforge.net">PASHA </a>at University of Cologne, Germany.</li>

<li>[<b>May 04, 2013</b>] I got invited to join the Program Committee of <a href="http://aces.snu.ac.kr/icpads13/Home.html">ICPADS 2013</a>.</li>


<li><b>[May 03, 2013</b>] The source code of <a href="http://cushaw2.sourceforge.net" target="_blank">CUSHAW2-GPU</a> has been released.</li>

<li>[<b>April 18, 2013</b>] The preprint paper for CUSHAW2 SOLiD color-space short-read alignment is available at <a href="http://arxiv.org/abs/1304.4766">arXiv</a>.</li>


<li>[<b>April 04, 2013</b>] CUDASW++ 3.0 paper has been published online by BMC Bioinformatics [<a href="http://www.biomedcentral.com/1471-2105/14/117">PDF</a>].</li>

<li>[<b>April 04, 2013</b>] <a href="http://cushaw2.sourceforge.net" target="_blank">CUSHAW2 v2.4</a> gets released. We have incorporated some new features in this release, described as follows.</li>
<ol>
<li>Different reference genome indexing from previous versions.</li>
<li>Support ABI SOLiD color-space single-end and paired-end/mate-paired alignment.</li>
<li>Improved base-space single-end and paired-end alignment quality.</li>
<li>Add a new option "-max_edit_dist" to allow users to specify the maximum edit distance of a reported alignment.</li>
<li>Each alignment in the SAM file has two tags given: NM (edit distance) and AS (alignment score).</li>
</ol>

<li>[<b>April 02, 2013</b>] <a href="https://sites.google.com/site/yongchaosoftware/mcuda-meme"> CUDA-MEME v3.0.7</a> gets released. This version can be compiled using CUDA 5.0 toolkit, and automatically detects the maximum number of resident threads per multiprocessor on GPUs.</li>

<li>[<b>March 15, 2013</b>] <a href="http://cudasw.sourceforge.net">CUDASW++ 3.0</a> paper gets accepted by BMC Bioinformatics.</li>

<li>[<b>March 12, 2013</b>] <a href="http://cushaw2.sourceforge.net">CUSHAW2-GPU</a> was first released.</li>
</ul>

</div>
</html>
