<html><head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta http-equiv="cache-control" content="max-age=0" />
<meta http-equiv="cache-control" content="no-cache" />
<meta http-equiv="expires" content="0" />
<meta http-equiv="expires" content="Tue, 01 Jan 1980 1:00:00 GMT" />
<meta http-equiv="pragma" content="no-cache" />
<link rel="stylesheet" type="text/css" href="style.css">
	<title>LESSA</title>
</head>
<body>
<!--header-->
<div id="header"></div>

<!--side bar-->
<div id="sidebar">

<!--photo-->
<div id="profile"></div>
<p style="text-align:center">[<a href="contact.html">Contact</a>][<a href="https://scholar.google.com/citations?user=qYQHl4sAAAAJ&hl=en&oi=ao">Google Scholar</a>]</p>

<!--navigator-->
<div id="nav">
<ul>
<li><a href="index.html">About Me</a></li>
<li><a href="news.html">What's New</a></li>
<li><a href="awards.html">Awards</a></li>
<li><a href="grants.html">Grants</a></li>
<li><a href="publications.html">Publications</a></li>
<li><a href="software.html">Software</a></li>
<li><a href="projects.html">Projects</a></li>
<li><a href="services.html">Services</a></li>
<li><a href="talks.html">Talks</a></li>
<li><a href="teaching.html">Teaching</a></li>
<li><a href="people.html">People</a></li>
<li><a href="contact.html">Contact</a></li>
</ul>
</div>

<!--last update-->
<p class="foot">
	<script language="JavaScript">
		document.write("Updated: " + document.lastModified);
	</script>
</p>

</div><!--sidebar-->

<div id="main">
<h1>LESSA: a <u>L</u>ibrary of Building Blocks for <u>E</u>fficient, <u>S</u>calable and <u>S</u>ervice-oriented <u>A</u>lgorithms</h1>
<hr>

<h2>1.Machine Learning Foundation Techniques and Applications</h2>
<p>We are dedicated to developing foundation techinques in machine learning, including but not limited to foundation models (e.g. large large models and large graph models), gradient desecent optimizers, distrbuted learning systems (e.g. deep learning and graph learning), deep learning compilers and automatic machine learning (AutoML), and further develop practical applications to solve real-world problems.</p>

<table>
<tr>
<td align="left" valign="middle" style="font-size: 11pt">
<p><h4>Example work:</h4></p>
<ul>
<li>Yue Jin, <u>Yongchao Liu</u>: 
	"<a href="http://2023.eurosys.org/docs/posters/eurosys23posters-final20.pdf" style="text-decoration:none">GPC: compiler-based optimization for sparse computations in graph neural networks</a>".
	The European Conference on Computer Systems (<b>EuroSys 2023</b>), 2023 (poster track).
</li>

<li>
Yun Yue, Zhiling Ye, Jiadi Jiang, <u>Yongchao Liu</u>, Ke Zhang:
	"<a href="https://openreview.net/forum?id=A954O4tDmU", style="text-decoration:none">AGD: an auto-switchable optimizer using stepwise gradient difference for preconditioning matrix</a>".
	37th Conference on Neural Information Processing Systems (<b>NeurIPS 2023</b>), 2023
</li>
	
<li>Yun Yue, Jiadi Jiang, Zhiling Ye, Gao Ning, <u>Yongchao Liu</u>, Ke Zhang:
	"<a href="https://dl.acm.org/doi/10.1145/3580305.3599501", style="text-decoration:none">Sharpness-aware minimization revisited: weighted sharpness as a regularization term</a>".
	2023 ACM SIGKDD International Conference on Knowledge Discovery and Data Mining (<b>KDD 2023</b>), 2023, pp. 3185–3194 (research track)
</li>

<li>
Yice Luo, Guannan Wang, <u>Yongchao Liu</u>, Jiaxin Yue, Weihong Cheng and Binjie Fei: 
	"<a href="https://dl.acm.org/doi/10.1145/3583780.3615477", style="text-decoration:none">FAF: a risk detection framework on industry-scale graphs</a>".
	32nd ACM International Conference on Information and Knowledge Management (<b>CIKM 2023</b>), 2023, pp. 4717–4723.
</li>

<li>Yuchen Zhou, Yanan Cao, <u>Yongchao Liu</u>, Yanmin Shang, Peng Zhang, Zheng Lin,
	Yun Yue, Baokun Wang, Xing Fu and Weiqiang Wang: "<a href="https://dl.acm.org/doi/10.1145/3543507.3583208" style="text-decoration:none">Multi-aspect heterogeneous graph augmentation</a>".
	The 2023 ACM Web Conference (<b>WWW 2023</b>), 2023, pp. 39-48.
</li>

<li>Houyi Li, Zhihong Chen, Chenliang Li, Rong Xiao, Hongbo Deng, Peng Zhang, <u>Yongchao Liu</u> and Haihong Tang: "<a href="https://dl.acm.org/doi/abs/10.1145/3404835.3462878" style="text-decoration:none">Path-based deep network for candidate item matching in recommenders</a>". 44th International ACM SIGIR Conference on Research and Development in Information Retrieval (<b>SIGIR 2021</b>), 2021, pp. 1493–1502.
</li>

<li>Yang Gao, Peng Zhang, Zhao Li, Chuan Zhou, Hong Yang, <u>Yongchao Liu</u>, and Yue Hu:
	"<a href="https://ieeexplore.ieee.org/document/9679011", style="text-decoration:none">Heterogeneous graph neural architecture search</a>".
	IEEE International Conference on Data Mining (<b>ICDM 2021</b>), 2021, pp. 1066-1071</li>

<li>Yun Yue, <u>Yongchao Liu</u>, Suo Tong, Minghao Li, Zhen Zhang, Chunyang Wen, Huanjun Bao, Lihong Gu, Jinjie Gu and Yixiang Mu:
	"<a href="https://2021.ecmlpkdd.org/wp-content/uploads/2021/07/sub_1040.pdf" style="text-decoration: none">Adaptive optimizers with sparse group lasso for neural networks in CTR prediction</a>".
	The European Conference on Machine Learning and Principles and Practice of Knowledge Discovery in Databases (<b>ECML-PKDD 2021</b>), 2021, pp. 314–329 .</li>

<li><u>Yongchao Liu</u>, Houyi Li, Guowei Zhang, Xintan Zeng, Yongyong Li, Bin Huang, Peng Zhang, Zhao Li, Xiaowei Zhu, Changhua He, Guowei Zhang, Xintan Zeng:
	"<a style="text-decoration: none", href="https://arxiv.org/abs/2104.10569">GraphTheta: A distributed graph neural network learning system with flexible training strategy</a>".
	arXiv:2104.10569, 2021
</li>
	
<li><u>Yongchao Liu</u>, Yue Jin, Yong Chen, Teng Teng, Hang Ou, Rui Zhao, Yao Zhang:
	"<a style="text-decoration:none" href="https://arxiv.org/pdf/2008.04567.pdf">Woodpecker-DL: Accelerating Deep Neural Networks via Hardware-Aware Multifaceted Optimizations</a>".
	arXiv:2008.04567 [cs.DC], 2020.
</li>
</ul>
</td>
</tr>
</table>

<h2>2. Compact Computing for Big Data</h2>
<p>As data volume increases exponentially and data compression is popularly used in data centers, I would expect that directly operating on compressed data would become commonplace in the future. However, parallel processing of compressed data is a challenging proposition for both shared-memory and distributed-memory systems and the challenges could come from constrained random accesses to uncompressed content, independent decompression of data blocks, balanced distribution of data, memory and computation, adaption of existing algorithms and applications to meet the requirements of compressed data processing and so on. Based on these concerns, I am conceiving of a new concept of computing and name it as <b>Compact Computing</b> tentatively. In general, Compact Computing targets robust, flexible and reproducible parallel processing of big data and consists of three core components, in principle: (1) tightly-coupled architectures, (2) compressive and elastic data representation, and (3) efficient, scalable and service-oriented algorithms and applications. In this context, component 1 can comprise conventional CPUs, a diversity of accelerators (e.g. FPGAs, GPUs, MIC processors and etc.) and fast interconnect communication facilities; component 2 concentrates on data structures and formats that enable efficient on-the-fly streaming compression and decompression; and component 3 targets the development of algorithms and applications that enable robust and efficient processing of big data streams in parallel. By centering around data, Compact Computing enables full-stack computation by tightly coupling algorithms with systems.</p>

<table>
<tr>
<td align="left" valign="middle" style="font-size: 11pt">
<p><h4>Example work:</h4></p>
<ul>
<li>Chengying Huan, <u>Yongchao Liu</u>, Heng Zhang, Shuaiwen Song, Santosh Pandey, Shiyang Chen, Xiangfei Fang, Yue Jin, Baptiste Lepers, Hang Liu, Yanjun Wu:
	"<a style="text-decoration:none", href="">TEA+: a novel temporal graph random walk engine with hybrid storage architecture</a>."
	ACM Transactions on Architecture and Code Optimization (<b>ACM TACO</b>), accepted, 2024.
</li>

<li>Yue Jin, Chengying Huan, Heng Zhang, <u>Yongchao Liu</u>, Shuaiwen Leon Song, Rui Zhao, Yao Zhang, Changhua He, Wenguang Chen:
	"<a href="https://www.computer.org/csdl/proceedings-article/pact/2023/425400a137/1TdF4jY5hkI", style="text-decoration:none">
	G-Sparse: compiler-driven acceleration for generalized sparse computation for graph neural networks on modern GPUs</a>".
	32nd International Conference on Parallel Architectures and Compilation Techniques (<b>PACT 2023</b>), 2023, pp. 137-149.
</li>
	
<li>Chengying Huan, Shuaiwen Leon Song, Santosh Pandey, Hang Liu, <u>Yongchao Liu</u>, Baptiste Lepers, Changhua He, Kang Chen, Jinlei Jiang, Yongwei Wu: 
	"<a href="https://dl.acm.org/doi/10.1145/3552326.3567491" style="text-decoration:none">TEA: a general-purpose temporal graph random walk engine</a>".
	The European Conference on Computer Systems (<b>EuroSys 2023</b>), 2023, pp. 182-198.
</li>

<li>Chengying Huan, Shuaiwen Leon Song, <u>Yongchao Liu</u>, Heng Zhang, Hang Liu, Charles He, Kang Chen, Jinlei Jiang, Yongwei Wu: 
	"<a href="https://dl.acm.org/doi/10.1145/3559009.3569648" style="text-decoration:none">T-GCN: a sampling based streaming graph neural network system with hybrid architecture</a>".
	31st International Conference on Parallel Architectures and Compilation Techniques (<b>PACT 2022</b>), 2022, pp. 69–82.
</li>

<li>Zhang H, Li L, Liu H, Zhuang D, Liu R, Huan C,  Song S, Tao D, <u>Liu Y</u>, He C, Wu Y, Song SL: "<a href="https://dl.acm.org/doi/10.1145/3524059.3532379" style="text-decoration:none">Bring orders into uncertainty: enabling efficient uncertain graph processing via novel path sampling on multi-accelerator system</a>". International Conference on Supercomputing 2022 (<b>ICS 2022</b>), 2022, 11:14.
</li>

<li>Huan C, Liu H, Liu M, <u>Liu Y</u>, He C, Chen K, Jiang J, Wu Y, Song SL: "<a href="https://ieeexplore.ieee.org/abstract/document/9835422", style="text-decoration:none">TeGraph: a novel general-purpose temporal graph computing engine</a>". 38th International Conference on Data Engineering (<b>ICDE 2022</b>), 2022, pp. 578-592.
</li>

<li><u>Yongchao Liu</u>, Houyi Li, Guowei Zhang, Xintan Zeng, Yongyong Li, Bin Huang, Peng Zhang, Zhao Li, Xiaowei Zhu, Changhua He, Guowei Zhang, Xintan Zeng:
	"<a style="text-decoration: none", href="https://arxiv.org/abs/2104.10569">GraphTheta: A distributed graph neural network learning system with flexible training strategy</a>".
	arXiv:2104.10569, 2021
</li>
	
<li><u>Liu Y</u>, Schmidt B: "<a href="http://link.springer.com/article/10.1007%2Fs11265-016-1216-4" style="text-decoration:none">LightSpMV: faster CUDA-compatible sparse matrix-vector multiplication using compressed sparse rows</a>". <b>Journal of Signal Processing Systems</b>, 2017, doi:10.1007/s11265-016-1216-4.
</li>
	
<li><u>Liu Y</u>, Hankeln T, Schmidt B: "<a href="http://dx.doi.org/10.1109/TCBB.2015.2430314" style="text-decoration:none">Parallel and space-efficient construction of Burrows-Wheeler transform and suffix array for big genome data</a>". <b>IEEE Transactions on Computational Biology and Bioinformatics</b>, 2016, 13(3): 592-598.
</li>
	
<li><u>Liu Y</u>, Schmidt B: "<a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=7245713" style="text-decoration:none">LightSpMV: faster CSR-based sparse matrix-vector multiplication on CUDA-enabled GPUs</a>". 26th IEEE International Conference on Application-specific Systems, Architectures and Processors (<b>ASAP 2015</b>), 2015, pp. 82-89 (<font color="#ff0000"><b>Best Paper Award</b>)</font>
</li>

<li><u>Liu Y</u>, Popp B, and Schmidt B: "<a href="http://www.ncbi.nlm.nih.gov/pubmed/24466273" style="text-decoration:none" target="_blank">CUSHAW3: sensitive and accurate base-space and color-space short-read alignment with hybrid seeding</a>." <b>PLOS ONE</b>, 2014, 9(1): e86869
</li>

<li><u>Liu Y</u>, Schmidt B: "<a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?tp=&amp;arnumber=6617698&amp;queryText%3DCUSHAW2-GPU%3A+empowering+faster+gapped+short-read+alignment+using+GPU+computing" style="text-decoration:none">CUSHAW2-GPU: empowering faster gapped short-read alignment using GPU computing</a>". <b>IEEE Design &amp; Test</b>, 2014, 31(1): 31-39
</li>

<li><u>Liu Y</u> and Schmidt B: "<a href="http://bioinformatics.oxfordjournals.org/content/28/18/i318.abstract" style="text-decoration:initial">Long read alignment based on maximal exact match seeds</a>". <b>Bioinformatics</b>, 2012, 28(18): i318-i324
</li>
	
<li><u>Liu Y</u>, Schmidt B, Maskell DL: "<a href="http://bioinformatics.oxfordjournals.org/content/28/14/1830.abstract" style="text-decoration:none">CUSHAW: a CUDA compatible short read aligner to large genomes based on the Burrows-Wheeler transform</a>". <b>Bioinformatics</b>, 2012, 28(14): 1830-1837
</li>
	
<li><u>Liu Y</u>, Schmidt B, Maskell DL: "<a href="http://www.biomedcentral.com/1471-2105/12/354" style="text-decoration:none">Parallelized short read assembly of large genomes using de Bruijn graphs</a>". <b>BMC Bioinformatics</b>, 2011, 12:354
</li>
	
</ul>
</td>
</tr>
</table>
	
<h2>3. Large-Scale Biological Sequence Analysis System</h2>
<p>
As part of my LESSA library, this project has been the core of my research on parallel and distributed algorithm design for bioinformatics, by employing a variety of tightly-coupled and loosely-coupled computing architectures, including heterogeneous computers with accelerators (e.g. Intel SSE, Intel AVX, Intel Xeon Phis, NVIDIA GPUs and AMD GPUs), cluster computing and cloud computing. My final objective is to establish an analysis system for large-scale biological sequences, in order to solve some critical and bottleneck problems in bioinformatics and computational biology, such as genome sequencing based on high-throughput sequencing technologies, meta-genomics, motif discovery, sequence alignment, and phylogenetic inference. Fig. 1 illustrates the diagram of an imaginaory biological data analysis system.
</p>
<table align="center" border="0" cellpadding="16">
<tr>
<td align="center" valign="middle"><img src="bio_system.png" height="auto" width="auto">
</tr>
<tr>
<td align="center" valign="middle">
<b>Fig. 1</b> System diagram for large-scale biological sequence analysis
</td>
</tr>
</table>

<tr>
<td align="left" valign="middle">
<p><h4>Example work:</h4> (The full lists of my software and publications are available <a href="https://yongchao-liu.github.io/publications.html">here</a>)</p>
<ul>
<li><u>Liu Y</u>, Maskell DL, Schmidt B: "<a href="http://www.biomedcentral.com/1756-0500/2/73" style="text-decoration:none">CUDASW++: optimizing Smith-Waterman sequence database searches for CUDA-enabled graphics processing units</a>". <b>BMC Research Notes</b>, 2009, 2:73 </li>

<li><u>Liu Y</u>, Schmidt B, Maskell DL: "<a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=5200019" style="text-decoration:none">MSA-CUDA: multiple sequence alignment on graphics processing units with CUDA</a>". 20th IEEE International Conference on Application-specific Systems, Architectures and Processors <b>(ASAP 2009</b>), 2009, 121-128 (<font color="#ff0000"><b>Best Paper Award</b>)</font></li>

<li><u>Liu Y</u>, Schmidt B, Maskell DL: "<a href="http://bioinformatics.oxfordjournals.org/content/26/16/1958.abstract" style="text-decoration:none">MSAProbs: multiple sequence alignment based on pair hidden Markov models and partition function posterior probabilities</a>". <b>Bioinformatics</b>, 2010, 26(16): 1958 -1964 </li>

<li><u>Liu Y</u>, Schmidt B, Liu W, Maskell DL: "<a href="http://www.sciencedirect.com/science/article/pii/S0167865509002840" style="text-decoration:none">CUDA-MEME: accelerating motif discovery in biological sequences using CUDA-enabled graphics processing units</a>". <b>Pattern Recognition Letters</b>, 2010, 31(14): 2170 - 2177</li>

<li><u>Liu Y</u>, Schmidt B, Douglas L. Maskell: "<a href="http://www.biomedcentral.com/1756-0500/3/93" style="text-decoration:none">CUDASW++2.0: enhanced Smith-Waterman protein database search on CUDA-enabled GPUs based on SIMT and virtualized SIMD abstractions</a>". <b>BMC Research Notes</b>, 2010, 3:93</li>

<li><u>Liu Y</u>, Schmidt B, Maskell DL: "<a href="http://www.biomedcentral.com/1471-2105/12/354" style="text-decoration:none">Parallelized short read assembly of large genomes using de Bruijn graphs</a>". <b>BMC Bioinformatics</b>, 2011, 12:354</li>

<li>Kuttippurathu L, Hsing M, <u>Liu Y</u>, Schmidt B, Maskell DL, Lee K, He A, Pu WT, and Kong SW: "<a href="http://bioinformatics.oxfordjournals.org/content/27/5/715.abstract" style="text-decoration:none">CompleteMOTIFs: DNA motif discovery platform for transcription factor binding experiments</a>". <b>Bioinformatics</b>, 2011, 27(5): 715-717</li>

<li><u>Liu Y</u>, Schmidt B, Maskell DL: "<a href="http://bioinformatics.oxfordjournals.org/content/28/14/1830.abstract" style="text-decoration:none">CUSHAW: a CUDA compatible short read aligner to large genomes based on the Burrows-Wheeler transform</a>". <b>Bioinformatics</b>, 2012, 28(14): 1830-1837</li>

<li><u>Liu Y</u> and Schmidt B: "<a href="http://bioinformatics.oxfordjournals.org/content/28/18/i318.abstract" style="text-decoration:none">Long read alignment based on maximal exact match seeds</a>". <b>Bioinformatics</b>, 2012, 28(18): i318-i324 (also from ECCB 2012)</li>

<li><u>Liu Y</u>, Schroeder J, Schmidt B: "<a href="http://bioinformatics.oxfordjournals.org/content/29/3/308.abstract" style="text-decoration:none">Musket: a multistage k-mer spectrum based error corrector for Illumina sequence data</a>". <b>Bioinformatics </b>, 2013, 29(3): 308-315</li>

<li><u>Liu Y</u>, Wirawan A, Schmidt B: "<a href="http://www.biomedcentral.com/1471-2105/14/117" style="text-decoration:none">CUDASW++ 3.0: accelerating Smith-Waterman protein database search by coupling CPU and GPU SIMD instructions</a>". <b>BMC Bioinformatics</b>, 2013, 14:117.</li>

<li><u>Liu Y</u>, Schmidt B: "<a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?tp=&amp;arnumber=6617698&amp;queryText%3DCUSHAW2-GPU%3A+empowering+faster+gapped+short-read+alignment+using+GPU+computing" style="text-decoration:none">CUSHAW2-GPU: empowering faster gapped short-read alignment using GPU computing</a>". <b>IEEE Design &amp; Test</b>, 2014, 31(1): 31-39</li>

<li>Ripp F, Krombholz CF, <u>Liu Y</u>, Weber M, Schaefer A, Schmidt B, Koeppel R, Hankeln T: "<a href="http://www.biomedcentral.com/1471-2164/15/639/abstract" style="text-decoration:none">All-Food-Seq (AFS): a quantifiable screen for species in biological samples by deep DNA sequencing</a>". <b>BMC Genomics</b>, 2014, 15:639</li>

<li><u>Liu Y</u>, Tran TT, Lauenroth F, Schmidt B: "<a href="http://dx.doi.org/10.1109/CLUSTER.2014.6968772" style="text-decoration:none">SWAPHI-LS: Smith-Waterman algorithm on Xeon Phi coprocessors for long DNA sequences</a>". 2014 IEEE International Conference on Cluster Computing <b>(Cluster 2014</b>), 2014, pp. 257-265 (<font color="#ff0000"><b>Best Paper Award Recommendation</b>)</font></li>

<li><u>Liu Y</u>, Schmidt B: "<a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=6868657" style="text-decoration:none">SWAPHI: Smith-Waterman protein database search on Xeon Phi coprocessors</a>". 25th IEEE International Conference on Application-specific Systems, Architectures and Processors (<b>ASAP 2014</b>), 2014, pp. 184-185</li>

<li><u>Liu Y</u>, Loewer M, Aluru S, Schmidt B: "<a href="http://dx.doi.org/10.1109/BIBM.2015.7359659" style="text-decoration:none">SNVSniffer: an integrated caller for germline and somatic SNVs based on Bayesian models</a>". 2015 IEEE International Conference on Bioinformatics and Biomedicine (<b>BIBM15</b>), 2015, pp. 83-90.</li>

<li><u>Liu Y</u>, Loewer M, Aluru S, Schmidt B: "<a style="text-decoration: none" href="http://bmcsystbiol.biomedcentral.com/articles/10.1186/s12918-016-0300-5">SNVSniffer: an integrated caller for germline and somatic single-nucleotide and indel mutations</a>". <b>BMC Systems Biology</b>, 2016, 10(Suppl 2):47</li>

<li>Pan T, Flick P, Jain C, <u>Liu Y</u> and Aluru S: "<a style="text-decoration:none" href="http://dl.acm.org/citation.cfm?id=2975211&CFID=870130274&CFTOKEN=58199232">Kmerind: A flexible parallel library for k-mer indexing of biological sequences on distributed memory systems</a>".  7th ACM Conference on Bioinformatics, Computational Biology, and Health Informatics (<b>ACM-BCB 2016</b>), 2016, pp. 422-433</li>

<li>Chan Y, Xu K, Lan H, Liu W, <u>Liu Y</u> and Schmidt B: ”<a style="text-decoration:none", href="https://doi.org/10.1109/IPDPS.2017.35">PUNAS: a parallel ungapped-alignment-featured seed verification for next-generation sequencing read alignment</a>”. 31st IEEE International Parallel and Distributed Processing Symposium (<b>IPDPS 2017</b>), 2017, pp. 52-61.</li>

<li>Lan H, Liu W, <u>Liu Y</u> and Schmidt B: ”<a style="text-decoration:none", href="https://doi.org/10.1109/IPDPS.2017.42">SWhybrid: a hybrid parallel framework for large-scale protein sequence database search</a>”. 31st IEEE International Parallel and Distributed Processing Symposium (<b>IPDPS 2017</b>), 2017, pp. 42-51.</li>

</ul>
</td>
</tr>
</table>
</div>
</html>
