<html><head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta http-equiv="cache-control" content="max-age=0" />
<meta http-equiv="cache-control" content="no-cache" />
<meta http-equiv="expires" content="0" />
<meta http-equiv="expires" content="Tue, 01 Jan 1980 1:00:00 GMT" />
<meta http-equiv="pragma" content="no-cache" />
<link rel="stylesheet" type="text/css" href="style.css">
	<title>Publications</title>
</head>

<body>
<!--header-->
<div id="header"></div>

<!--side bar-->
<div id="sidebar">

<!--photo-->
<div id="profile"></div>
<p style="text-align:center">[<a href="contact.html">Contact</a>][<a href="https://scholar.google.com/citations?user=qYQHl4sAAAAJ&hl=en&oi=ao">Google Scholar</a>]</p>

<!--navigator-->
<div id="nav">
<ul>
<li><a href="index.html">About Me</a></li>
<li><a href="news.html">What's New</a></li>
<li><a href="awards.html">Awards</a></li>
<li><a href="grants.html">Grants</a></li>
<li><a href="publications.html">Publications</a></li>
<li><a href="software.html">Software</a></li>
<li><a href="projects.html">Projects</a></li>
<li><a href="services.html">Services</a></li>
<li><a href="talks.html">Talks</a></li>
<li><a href="teaching.html">Teaching</a></li>
<li><a href="people.html">People</a></li>
<li><a href="contact.html">Contact</a></li>
</ul>
</div>

<!--last update-->
<p class="foot">
	<script language="JavaScript">
		document.write("Updated: " + document.lastModified);
	</script>
</p>

</div><!--sidebar-->

<div id="main">
<h1>Publications</h1>
<hr>
<h2>Patents Granted</h2>
<ol>
<li>Qiyin Huang, <u>Yongchao Liu</u>, Haitao Zhang, Chengping Yang: "<a href="https://patents.google.com/patent/TW202101305A/en">Method, device and equipment for updating convolutional neural network using GPU cluster</a>", 01/2021.</li>
<li>Haitao Zhang, <u>Yongchao Liu</u>: "<a href="https://patents.google.com/patent/US10891517B2/en?inventor=yongchao+liu&assignee=alibaba&oq=yongchao+liu+alibaba">Vehicle accident image processing method and apparatus</a>", 01/2021.</li>
<li><u>Yongchao Liu</u>, Sizhong Li, Guozhen Pan, Jianguo Xu, Qiyin Huang: "<a href="https://patents.google.com/patent/CN110033091B/en">Model-based prediction method and device</a>", 09/2020. </li>
<li><u>Yongchao Liu</u>, Haitao Zhang, Yufeng Guo: "<a href="https://patents.google.com/patent/CN108921811B/en">Method and device for detecting damage of article and article damage detector</a>, 06/2020.</li>
<li>Guozhen Pan, Jianguo Xu, <u>Yongchao Liu</u> Haitao Zhang, Qiyin Huang, Guanyin Zhu: "<a href="https://patents.google.com/patent/CN109359732B">Chip and data processing method based on chip</a>", 04/2020.</li>
<li><u>Yongchao Liu</u> Qiyin Huang, Guozhen Pan, Sizhong Li, Jianguo Xu, Haitao Zhang, Lin Wang: "<a href="https://patents.google.com/patent/US20200134400A1/en">Fast computation of a convolutional neural network</a>", 04/2020</li>
</ol>
	
<h2>Book Chapters</h2>
<ol>
<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://www.springer.com/us/book/9783319598246" style="text-decoration:none">CUSHAW Suite:  parallel and efficient algorithms for NGS read alignment</a>". <b>Algorithms for Next-Generations Sequencing Data: Techniques, Approaches and Applications</b>, edited by Mourad Elloumi, Springer, 2017.</li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://www.sciencedirect.com/science/article/pii/B9780128038192000070" style="text-decoration:none">Pairwise DNA sequence alignment optimization</a>". <b>High Performance Parallelism Pearls Volume Two - Multicore and Many-core Programming Approaches</b>, edited by James Reinders and Jim Jeffers, 2015, pp. 43-54</li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://link.springer.com/protocol/10.1007/978-1-62703-646-7_14" style="text-decoration:none">Multiple protein sequence alignment with MSAProbs</a>". <b>Methods in Molecular Biology</b>, Edited by David Russell, Springer, 2014, 1079: 211-218</li>

<li>Lukasz Ligowski, Witold Rudnicki, <u>Yongchao Liu</u>, and Bertil Schmidt: "<a href="http://www.elsevierdirect.com/ISBN/9780123849885/GPU-Computing-Gems-Emerald-Edition" style="text-decoration:none">Accurate scanning of sequence databases with the Smith-Waterman algorithm</a>".&nbsp;<b>GPU Computing Gems</b>, Edited by Wen-mei W. Hwu, Elsevier 2011, Pages 155-172</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt , and Douglas L. Maskell: "<a href="http://www.crcpress.com/product/isbn/9781439814888" style="text-decoration:none">Parallel bioinformatics algorithms for CUDA-enabled GPUs</a>".<b> Bioinformatics: High Performance Parallel Computer Architectures</b>, Edited by Bertil Schmidt, CRC Press 2010, Pages 117-137</li>

</ol>

<h2>Journal Articles</h2>
<ol>
<li>Xiaowei Zhu, Zhisong Fu, Zhenxuan Pan, Jin Jiang, Chuntao Hong, <u>Yongchao Liu</u>, Yang Fang, Wenguang Chen, Changhua He: "<a style="text-decoration:none", href="https://dl.acm.org/doi/10.1145/3469379.3469389">Taking the Pulse of Financial Activities with Online Graph Processing</a>". <b>ACM SIGOPS Operating Systems Review</b>, vol 55, no.1, 2021</li>
	
<li>Tony C Pan, Patrick Flick, Chirag Jain, <u>Yongchao Liu</u> and Srinivas Aluru: "Kmerind: a exible parallel library for k-mer indexing of biological sequences on distributed memory systems". <b>IEEE/ACM Transactions on Computational Biology and Bioinformatics</b> (<i>impact factor</i> 1.955), 2019, 16(4):1117-1131. </li>
	
<li>Shuozhi Xu, Thomas Payne, Hao Chen, <u>Yongchao Liu</u>, Liming Xiong, Youping Chen and David McDowell: "<a href="
https://doi.org/10.1557/jmr.2018.8" style="text-decoration: none">PyCAC: The concurrent atomistic-continuum simulator with a Python scripting interface</a>". <b>Journal of Materials Research</b> (<i>impact factor</i> 1.673), 2018, 33(7):857-871. </li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://link.springer.com/article/10.1007%2Fs11265-016-1216-4" style="text-decoration:none">LightSpMV: faster CUDA-compatible sparse matrix-vector multiplication using compressed sparse rows</a>". <b>Journal of Signal Processing Systems</b> (<i>impact factor</i> 0.508), 2018, 90(1):69-86. </li>

<li><u>Yongchao Liu</u>, Fabian Ripp, Rene Koeppe, Hanno Schmidt, Lukas Hellmann, Mathias Weber, Christopher Felix Krombholz, Bertil Schmidt and Thomas Hankeln: "<a href="https://doi.org/10.1093/bioinformatics/btw822" style="text-decoration:none">AFS: identification and quantification of species composition by metagenomic sequencing"</a>. <b>Bioinformatics</b> (<i>impact factor</i> 7.307), 2017, 33 (9): 1396-1398.</li>

<li>Sharma V. Thankachan, Sriram P. Chockalingam, <u>Yongchao Liu</u>, Ambujam Krishnan, Srinivas Aluru: "<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1658-0" style="text-decoration:none">A greedy alignment-free distance estimator for phylogenetic inference</a>". <b>BMC Bioinformatics</b> (<i>impact factor</i> 2.58), 2017, 18(Suppl 8):238.</li>

<li>Jorge Gonzalez-Dominguez, <u>Yongchao Liu</u>, Juan Tourino and Bertil Schmidt: "<a href="http://bioinformatics.oxfordjournals.org/content/32/24/3826" style="text-decoration: none">MSAProbs-MPI: parallel multiple sequence aligner for distributed-memory systems</a>". <b>Bioinformatics</b> (<i>impact factor</i> 5.766), 2016, 32(24): 3826-3828</li>

<li><u>Yongchao Liu</u>, Martin Loewer, Srinivas Aluru and Bertil Schmidt: "<a style="text-decoration:none" href="http://bmcsystbiol.biomedcentral.com/articles/10.1186/s12918-016-0300-5">SNVSniffer: an integrated caller for germline and somatic single-nucleotide and indel mutations</a>". <b>BMC Systems Biology</b> (<i>impact factor</i> 2.435), 2016, 10(suppl 2): 47</li>

<li>Jorge González-Domínguez, <u>Yongchao Liu</u>, Bertil Schmidt: "<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0145490" style="text-decoration: none">Parallel and scalable short-read alignment on multi-core clusters using UPC++</a>". <b>PLoS One</b> (<i>impact factor</i> 3.234), 2016, 11(1): e0145490.</li>

<li>Tuan Tu Tran, <u>Yongchao Liu</u>, Bertil Schmidt: "<a style="text-decoration:none" href="http://dx.doi.org/10.1016/j.parco.2015.11.001">Bit-parallel approximate pattern matching: Kepler GPU versus Xeon Phi</a>". <b>Parallel Computing</b> (<i>impact factor</i> 1.511), 2016, 54: 128-138.</li>

<li>Sharma V. Thankachan, Sriram P. Chockalingam, <u>Yongchao Liu</u>, Alberto Apostolico and Srinivas Aluru: "<a style="text-decoration:none" href="http://online.liebertpub.com/doi/10.1089/cmb.2015.0217">ALFRED: a practical method for alignment-free distance computation</a>". <b>Journal of Computational Biology</b> (<i>impact factor</i> 1.737), 2016, 23(6): 452-460</li>

<li><u>Yongchao Liu</u>, Thomas Hankeln, and Bertil Schmidt: "<a href="http://dx.doi.org/10.1109/TCBB.2015.2430314" style="text-decoration:none">Parallel and space-efficient construction of Burrows-Wheeler transform and suffix array for big genome data</a>". <b>IEEE Transactions on Computational Biology and Bioinformatics</b> (<i>impact factor</i> 1.536), 2016, 13(3): 592-598.</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt: "<a href="http://onlinelibrary.wiley.com/doi/10.1002/cpe.3371/full" style="text-decoration:none">GSWABE: faster GPU-accelerated sequence alignment with optimal alignment retrieval for short DNA sequences</a>". <b>Concurrency and Computation: Practice and Experience</b> (<i>impact factor</i> 0.784), 2015, 27: 958-972</li>


<li>Fabian Ripp, Christopher F Krombholz, <u>Yongchao Liu</u>, Mathias Weber, Anne Sch?fer, Bertil Schmidt, Rene K?ppel and Thomas Hankeln: "<a href="http://www.biomedcentral.com/1471-2164/15/639/abstract" style="text-decoration:none">All-Food-Seq (AFS): a quantifiable screen for species in biological samples by deep DNA sequencing</a>". <b>BMC Genomics</b> (<i>impact factor</i> 4.40), 2014, 15:639 [<font color="#ff0000"><b>Reported by <a href="http://www.ausfoodnews.com.au/2013/03/28/new-dna-traceability-test-will-improve-food-composition-screening.html">Austrilian Food News</a></b></font>]</li>

<li>Adrianto Wirawan, Robert S Harris, <u>Yongchao Liu</u>, Bertil Schmidt and Jan Schr?der: "<a href="http://www.biomedcentral.com/1471-2105/15/131/abstract" style="text-decoration:none">HECTOR: A parallel multistage homopolymer spectrum based error corrector for 454 sequencing data</a>." <b>BMC Bioinformatics</b> (<i>impact factor</i> 3.02), 2014, 15:131</li>

<li><u>Yongchao Liu</u>, Bernt Popp, and Bertil Schmidt: "<a href="http://www.ncbi.nlm.nih.gov/pubmed/24466273" style="text-decoration:none" target="_blank">CUSHAW3: sensitive and accurate base-space and color-space short-read alignment with hybrid seeding</a>." <b>PLOS ONE</b> (<i>impact factor</i> 3.730), 2014, 9(1): e86869</li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?tp=&amp;arnumber=6617698&amp;queryText%3DCUSHAW2-GPU%3A+empowering+faster+gapped+short-read+alignment+using+GPU+computing" style="text-decoration:none">CUSHAW2-GPU: empowering faster gapped short-read alignment using GPU computing</a>". <b>IEEE Design &amp; Test</b> (<i>impact factor</i> 1.623), 2014, 31(1): 31-39</li>

<li><u>Yongchao Liu</u>, Adrianto Wirawan and Bertil Schmidt: "<a href="http://www.biomedcentral.com/1471-2105/14/117" style="text-decoration:none">CUDASW++ 3.0: accelerating Smith-Waterman protein database search by coupling CPU and GPU SIMD instructions</a>". <b>BMC Bioinformatics </b>(<i>impact factor</i> 3.02), 2013, 14:117.</li>

<li><u>Yongchao Liu</u>, Jan Schr?der, and Bertil Schmidt: "<a href="http://bioinformatics.oxfordjournals.org/content/29/3/308.abstract" style="text-decoration:none">Musket: a multistage k-mer spectrum based error corrector for Illumina sequence data</a>". <b>Bioinformatics </b>(<i>impact factor</i> 5.323), 2013, 29(3): 308-315</li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://bioinformatics.oxfordjournals.org/content/28/18/i318.abstract" style="text-decoration:none">Long read alignment based on maximal exact match seeds</a>". <b>Bioinformatics</b>(<i>impact factor</i> 5.468), 2012, 28(18): i318-i324 (also from ECCB 2012)</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, and Douglas L. Maskell: "<a href="http://bioinformatics.oxfordjournals.org/content/28/14/1830.abstract" style="text-decoration:none">CUSHAW: a CUDA compatible short read aligner to large genomes based on the Burrows-Wheeler transform</a>". <b>Bioinformatics</b> (<i>impact factor </i>5.468), 2012, 28(14): 1830-1837</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, and Douglas L. Maskell: "<a href="http://www.biomedcentral.com/1471-2105/12/354" style="text-decoration:none">Parallelized short read assembly of large genomes using de Bruijn graphs</a>". <b>BMC Bioinformatics</b> (<i>impact factor</i> 3.43), 2011, 12:354</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, and Douglas L. Maskell: "<a href="http://www.biomedcentral.com/1471-2105/12/85" style="text-decoration:none">DecGPU: distributed error correction on massively parallel graphics processing units using CUDA and MPI</a>". <b>BMC Bioinformatics</b> (<i>impact factor</i> 3.43), 2011, 12:85 [<font color="#ff0000"><b>Reported by <a href="https://www.genomeweb.com/informatics/next-gen-gpus">GenomeWeb</a></b></font>] </li>

<li>Lakshmi Kuttippurathu, Michael Hsing, <u>Yongchao Liu</u>, Bertil Schmidt, Douglas L.Maskell, Kyungjoon Lee, Aibin He, William T. Pu, and Sek Won Kong: "<a href="http://bioinformatics.oxfordjournals.org/content/27/5/715.abstract" style="text-decoration:none">CompleteMOTIFs: DNA motif discovery platform for transcription factor binding experiments</a>". <b>Bioinformatics </b>(<i>impact factor </i>4.926), 2011, 27(5): 715-717</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, and Douglas L. Maskell: "<a href="http://bioinformatics.oxfordjournals.org/content/26/16/1958.abstract" style="text-decoration:none">MSAProbs: multiple sequence alignment based on pair hidden Markov models and partition function posterior probabilities</a>". <b>Bioinformatics</b> (<i>impact factor </i>4.926), 2010, 26(16): 1958 -1964 </li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, and Douglas L. Maskell: "<a href="http://www.biomedcentral.com/1756-0500/3/93" style="text-decoration:none">CUDASW++2.0: enhanced Smith-Waterman protein database search on CUDA-enabled GPUs based on SIMT and virtualized SIMD abstractions</a>". <b>BMC Research Notes</b>, 2010, 3:93</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, Weiguo Liu, and Douglas L. Maskell: "<a href="http://www.sciencedirect.com/science/article/pii/S0167865509002840" style="text-decoration:none">CUDA-MEME: accelerating motif discovery in biological sequences using CUDA-enabled graphics processing units</a>". <b>Pattern Recognition Letters</b> (<i>impact factor </i>1.559), 2010, 31(14): 2170 - 2177</li>

<li><u>Yongchao Liu</u>, Douglas L. Maskell, and Bertil Schmidt: "<a href="http://www.biomedcentral.com/1756-0500/2/73" style="text-decoration:none">CUDASW++: optimizing Smith-Waterman sequence database searches for CUDA-enabled graphics processing units</a>". <b>BMC Research Notes</b>, 2009, 2:73</li>


<li>李庆诚, <u>刘永超(Yongchao Liu)</u>, 刘嘉鑫: "<a href="http://d.wanfangdata.com.cn/periodical_jsjyy2007z1109.aspx" style="text-decoration:initial">平台无关的PDF嵌入式高性价比解析器设计与实现(Design and implementation of a platform-independent, high-performance-cost-ratio, embedded PDF format parser<sup>*</sup>)</a>". <b>计算机应用</b>(Computer Applications), 2007, 27(z1). [<sup>*</sup>translated by myself]</li>

</ol>


<h2>Conference/Workshop Papers</h2>
<ol>
<li>Heng Zhang, Lingda Li, Hang Liu, Donglin Zhuang, Rui Liu, Chengying Huan, Shuang Song, Dingwen Tao, Yongchao Liu, Changhua He, Yanjun Wu, Shuaiwen Leon Song: "Bring orders into uncertainty: enabling efficient uncertain graph processing via novel path sampling on multi-accelerator system". International Conference on Supercomputing 2022 (<b>ICS 2022</b>), June 2022, in press.</li>

<li>Chengying Huan, Hang Liu, Mengxing Liu, <u>Yongchao Liu</u>, Changhua He, Kang Chen, Jinlei Jiang, Yongwei Wu, Shuaiwen Leon Song: "TeGraph: a novel general-purpose temporal graph computing engine". 38th International Conference on Data Engineering (<b>ICDE 2022</b>), May, 2022.</li>

<li>Yang Gao, Peng Zhang, Zhao Li, Chuan Zhou, Hong Yang, <u>Yongchao Liu</u>, and Yue Hu: "<a href="https://ieeexplore.ieee.org/document/9679011", style="text-decoration:none">Heterogeneous graph neural architecture search</a>". IEEE International Conference on Data Mining (<b>ICDM 2021</b>), 2021, pp. 1066-1071</li>
	
<li>Yun Yue, <u>Yongchao Liu</u>, Suo Tong, Minghao Li, Zhen Zhang, Chunyang Wen, Huanjun Bao, Lihong Gu, Jinjie Gu and Yixiang Mu: "<a href="https://2021.ecmlpkdd.org/wp-content/uploads/2021/07/sub_1040.pdf" style="text-decoration: none">Adaptive optimizers with sparse group lasso for neural networks in CTR prediction</a>". The European Conference on Machine Learning and Principles and Practice of Knowledge Discovery in Databases (<b>ECML-PKDD 2021</b>), 2021, pp. 314–329 .</li>
	
<li>Houyi Li, Zhihong Chen, Chenliang Li, Rong Xiao, Hongbo Deng, Peng Zhang, <u>Yongchao Liu</u> and Haihong Tang: "<a href="https://dl.acm.org/doi/abs/10.1145/3404835.3462878" style="text-decoration:none">Path-based deep network for candidate item matching in recommenders</a>". 44th International ACM SIGIR Conference on Research and Development in Information Retrieval (<b>SIGIR 2021</b>), 2021, pp. 1493–1502.</li>
	
<li> Yue Jin, <u>Yongchao Liu</u>, Yong Chen, Rui Zhao, Yao Zhang: "<a href="https://on-demand-gtc.gputechconf.com/gtcnew/sessionview.php?sessionName=cns20774-%E5%9F%BA%E4%BA%8E%E6%A8%A1%E5%9E%8B%E7%9A%84+gpu+%E6%80%A7%E8%83%BD%E4%BC%B0%E7%AE%97%E5%8F%8A%E5%85%B6%E5%9C%A8%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0%E8%BF%90%E7%AE%97%E6%80%A7%E8%83%BD%E4%BC%98%E5%8C%96%E4%B8%8A%E7%9A%84%E5%BA%94%E7%94%A8" style="text-decoration:none">Model-based cost estimation and its application in deep learning operation optimizations</a>". GPU Technology Conference 2020 (<b>GTC 2020</b>), 2020, China (talk given by Yue Jin).</li>
	
<li><u>Yongchao Liu</u>, Yue Jin, Yong Chen, Teng Teng, Hang Ou, Rui Zhao, Yao Zhang: "<a href="https://on-demand.gputechconf.com/gtc-cn/2019/pdf/CN9274/presentation.pdf", style="text-decoration:none">Woodpecker-DL: an efficient compiler for accelerating deep learning on heterogeneous computing architectures</a>". GPU Technology Conference 2019 (<b>GTC 2019</b>), 2019, China (talk given by Yong Chen).</li>
	
<li>Shuozhi Xu, Thomas G. Payne, Hao Chen, <u>Yongchao Liu</u>, Liming Xiong, Youping Chen, David L. McDowell: "<a href="http://www.programmaster.org/PM/PM.nsf/ApprovedAbstracts/4BF727B3A26116D685258146006D5E0A?OpenDocument", style="text-decoration:none">PyCAC: the concurrent atomistic-continuum simulation environment</a>”. 2018 TMS Annual Meeting &amp; Exhibition (<b>TMS 2018</b>)</li>

<li>Yuandong Chan, Kai Xu, Haidong Lan, Weiguo Liu, <u>Yongchao Liu</u> and Bertil Schmidt: ”<a style="text-decoration:none" href="https://doi.org/10.1109/IPDPS.2017.35">PUNAS: a parallel ungapped-alignment-featured seed verification for next-generation sequencing read alignment</a>”. 31st IEEE International Parallel and Distributed Processing Symposium (<b>IPDPS 2017</b>), 2017, pp. 52-61.</li>

<li>Haidong Lan, Weiguo Liu, <u>Yongchao Liu</u> and Bertil Schmidt: ”<a style="text-decoration:none" href="https://doi.org/10.1109/IPDPS.2017.42">SWhybrid: a hybrid parallel framework for large-scale protein sequence database search</a>”. 31st IEEE International Parallel and Distributed Processing Symposium (<b>IPDPS 2017</b>), 2017, pp. 42-51.</li>

<li>Tony Pan, Patrick Flick, Chirag Jain, <u>Yongchao Liu</u> and Srinivas Aluru: "<a style="text-decoration:none" href="http://dl.acm.org/citation.cfm?id=2975211&CFID=870130274&CFTOKEN=58199232">Kmerind: A flexible parallel library for k-mer indexing of biological sequences on distributed memory systems</a>".  7th ACM Conference on Bioinformatics, Computational Biology, and Health Informatics (<b>ACM-BCB 2016</b>), 2016, pp. 422-433</li>

<li><u>Yongchao Liu</u>, Tony Pan, and Srinivas Aluru: "<a style="text-decoration: none" href="https://doi.org/10.1109/SBAC-PAD.2016.26">Parallel pairwise correlation computation on Intel Xeon Phi clusters</a>". 28th International Symposium on Computer Architecture and High Performance Computing (<b>SBAC-PAD 2016</b>), 2016, pp. 141-149 </li>


<li><u>Yongchao Liu</u>, Martin Loewer, Srinivas Aluru, Bertil Schmidt: "<a href="http://dx.doi.org/10.1109/BIBM.2015.7359659" style="text-decoration:none">SNVSniffer: an integrated caller for germline and somatic SNVs based on Bayesian models</a>". 2015 IEEE International Conference on Bioinformatics and Biomedicine (<b>BIBM 2015</b>), 2015, pp. 83-90.</li>

<li>Sharma V. Thankachan, Sriram P. Chockalingam, <u>Yongchao Liu</u>, Ambujam Krishnan, Srinivas Aluru: "<a href="http://dx.doi.org/10.1109/ICCABS.2015.7344711" style="text-decoration:none">A greedy alignment-free distance estimator for phylogenetic inference (extended abstract)</a>". 5th IEEE International Conference on Computational Advances in Bio and Medical Sciences (<b>ICCABS 2015</b>), 2015, pp. 1-1.</li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a style="text-decoration: none" href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=7245713">LightSpMV: faster CSR-based sparse matrix-vector multiplication on CUDA-enabled GPUs</a>". 26th IEEE International Conference on Application-specific Systems, Architectures and Processors (<b>ASAP 2015</b>), 2015, pp. 82-89 (<font color="#ff0000"><b>Best Paper Award</b>)</font></li>

<li><u>Yongchao Liu</u>, Jorge González-Domínguez, Bertil Schmidt: "<a href="http://on-demand.gputechconf.com/gtc/2015/posters/GTC_2015_Developer_Algorithms_11_P5267_WEB.pdf" style="text-decoration:none">Faster compressed sparse row (CSR)-based sparse matrix-vector multiplication using CUDA</a>". GPU Technology Conference 2015 (<b>GTC 2015</b>), San Jose, USA, 2015</li>

<li>Tuan Tu Tran, Simon Schindel, <u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=6970650" style="text-decoration:none">Bit-Parallel approximate pattern matching on the Xeon Phi coprocessor</a>". 26th International Symposium on Computer Architecture and High Performance Computing (<b>SBAC-PAD 2014</b>), 2014, pp. 81-88</li>

<li><u>Yongchao Liu</u>, Tuan-Tu Tran, Felix Lauenroth and Bertil Schmidt: "<a href="http://dx.doi.org/10.1109/CLUSTER.2014.6968772" style="text-decoration:none">SWAPHI-LS: Smith-Waterman algorithm on Xeon Phi coprocessors for long DNA sequences</a>". 2014 IEEE International Conference on Cluster Computing <b>(Cluster 2014</b>), 2014, pp. 257-265 (<font color="#ff0000"><b>Best Paper Award Recommendation</b>)</font></li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=6868657" style="text-decoration:none">SWAPHI: Smith-Waterman protein database search on Xeon Phi coprocessors</a>". 25th IEEE International Conference on Application-specific Systems, Architectures and Processors (<b>ASAP 2014</b>), 2014, pp. 184-185 [<a href="http://arxiv.org/abs/1404.4152" style="text-decoration:none">full version at arXiv</a>] </li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://on-demand.gputechconf.com/gtc/2014/poster/pdf/P4117_CUSHAW_sequencing_CUDA_GPU.pdf" style="text-decoration:none">CUSHAW Software Package: harnessing CUDA-enabled GPUs for next generation sequencing read alignment</a>". GPU Technology Conference 2014 (<b>GTC 2014</b>), San Jose, USA, 2014
</li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://link.springer.com/chapter/10.1007%2F978-3-642-55195-6_23" style="text-decoration:none">Faster GPU-accelerated Smith-Waterman algorithm with alignment backtracking for short DNA sequences</a>". 10th International Conference on Parallel Processing and Applied Mathematics (<b>PPAM 2013</b>), appear in Lecture Notes in Computer Science 8385, pp. 247-257</li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: "<a href="http://bioinformatics.oxfordjournals.org/content/28/18/i318.abstract" style="text-decoration:initial">Long read alignment based on maximal exact match seeds</a>". 11th European Conference on Computational Biology (<b>ECCB 2012</b>), Basel, Switzerland (also published in the <i>Bioinformatics </i>journal)</li>

<li><u>Yongchao Liu</u> and Bertil Schmidt: <a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?reload=true&amp;arnumber=6270707&amp;contentType=Conference+Publications" style="text-decoration:none"> Evaluation of GPU-based seed generation for computational genomics using Burrows-Wheeler transform</a>". 26th IEEE International Symposium on Parallel and Distributed Processing Workshops and Phd Forum (<b>IPDPSW 2012</b>), 684-690</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, and Douglas L. Maskell: "<a href="http://on-demand.gputechconf.com/gtc/2012/posters/P0457_GTC_2012_poster.pdf" style="text-decoration:none">A fast CUDA compatible short read aligner to large genomes</a>". GPU Technology Conference 2012 (<b>GTC 2012</b>), San Jose, USA, 2012</li>

<li>Weiguo Liu, Bertil Schmidt, <u>Yongchao Liu</u>, and Wolfgang Müller-Wittig: "<a href="http://ieeexplore.ieee.org/xpl/freeabs_all.jsp?arnumber=6121283" style="text-decoration:none">Mapping of the BLASTP algorithm onto GPU clusters</a>". 17th IEEE International Conference on Parallel and Distributed Systems (<b>ICPADS 2011</b>), 2011, 236-243</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, and Douglas L. Maskell: "<a href="http://www.hicomb.org/papers/HICOMB2011-03.pdf" style="text-decoration:none">An ultrafast scalable many-core motif discovery algorithm for multiple GPUs</a>". 25th IEEE International Symposium on Parallel and Distributed Processing Workshops and Phd Forum (<b>IPDPSW 2011</b>), 428-434</li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, and Douglas L. Maskell: "<a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=5200019" style="text-decoration:none">MSA-CUDA: multiple sequence alignment on graphics processing units with CUDA</a>". 20th IEEE International Conference on Application-specific Systems, Architectures and Processors <b>(ASAP 2009</b>), 2009, 121-128 (<font color="#ff0000"><b>Best Paper Award</b>)</font></li>

<li><u>Yongchao Liu</u>, Bertil Schmidt, and Douglas L. Maskell: "<a href="http://www.hicomb.org/papers/HICOMB2009-02.pdf" style="text-decoration:none">Parallel reconstruction of neighbor-Joining trees for large multiple sequence alignments using CUDA</a>". 23th IEEE International Symposium on Parallel and Distributed Processing (<b>IPDPS 2009</b>), 2009, 1-8.</li>

</ol>

<h2>Technical Reports</h2>
<ol>
<li>Qinkai Zheng, Houyi Li, Peng Zhang, Zhixiong Yang, Guowei Zhang, Xintan Zeng, <u>Yongchao Liu</u>: "<a style="text-decoration:none", href="">GIPA: general information propagation algorithm for graph learning</a>. https://arxiv.org/abs/2105.06035, 2021</li>
<li>Houyi Li, <u>Yongchao Liu</u>, Yongyong Li, Bin Huang, Peng Zhang, Guowei Zhang, Xintan Zeng, Kefeng Deng, Wenguang Chen, Changhua He: "<a style="text-decoration: none", href="https://arxiv.org/abs/2104.10569">GraphTheta: A distributed graph neural network learning system with flexible training strategy</a>". arXiv:2104.10569, 2021</li>
<li><u>Yongchao Liu</u>, Yue Jin, Yong Chen, Teng Teng, Hang Ou, Rui Zhao, Yao Zhang: "<a style="text-decoration:none" href="https://arxiv.org/pdf/2008.04567.pdf">Woodpecker-DL: Accelerating Deep Neural Networks via Hardware-Aware Multifaceted Optimizations</a>". arXiv:2008.04567 [cs.DC], 2020.</li>
<li><u>Yongchao Liu</u>, Tony Pan, Oded Green and Srinivas Aluru: "<a style="text-decoration:none" href="http://arxiv.org/abs/1704.03767">Parallelized Kendall's tau coefficient computation via SIMD vectorized sorting on many-integrated-core processors</a>". arXiv:1704.03767, 2017. </li>

<li><u>Yongchao Liu</u> and Srinivas Aluru: "<a style="text-decoration:none" href="http://arxiv.org/abs/1604.04815">LightScan: faster scan primitive on CUDA compatible manycore processors</a>". arXiv:1604.04815 [Cs.DC], 2016.</li>

<li><u>Yongchao Liu</u>: "<a style="text-decoration:none" href="https://www.researchgate.net/publication/309672981_Research_Statement_for_Assistant_Professorship_Application_2016">Research statement for assistant professorship application</a>". <b>ResearchGate</b>, 2016, doi: 10.13140/RG.2.2.28828.90243/1 </li>

<li><u>Yongchao Liu</u>: "<a style="text-decoration:none" href="https://www.researchgate.net/publication/309673143_OpenGraphAssembly_abstract_modularize_and_parallelize_fundamental_building_blocks_for_graph-based_genome_assembly">OpenGraphAssembly: abstract, modularize and parallelize fundamental building blocks for graph-based genome assembly</a>". <b>ResearchGate</b>, 2015, doi: 10.13140/RG.2.2.11212.825</li>

<li><u>Yongchao Liu</u>, Bernt Popp, and Bertil Schmidt: "<a href="http://arxiv.org/abs/1304.4766" style="text-decoration:none">High-speed and accurate color-space short-read alignment with CUSHAW2</a>". arXiv:1304.4766 [q-bio.GN], 2013</li>

</ol>

</div>
</html>
